docs/001-ci-strategy.adoc

```diff
@@ -0,0 +1,101 @@
+= ADR 001 - Unit and Integration Testing CI Strategy
+Philipp Hoenisch <philipp@10101.finance>
+3.0, July 29, 2022: AsciiDoc article template
+:toc:
+:icons: font
+:attributes: 2023-01-17
+
+This document outlines the reasoning behind our CI strategy.
+
+== Decision
+
+* We have 2 stages of tests
+** Unit tests:
+In unit tests we stub all DB/io/networking/etc to achieve a sub *10 minute* runtime.
+Unit tests can run on every commit.
+** Integration tests:
+Integration tests run only occasionally and *only* if all unit tests have passed.
+We run integration tests only on `bors r+`.
```

Comment:
In my opinion, running tests after the change has been merged is cumbersome. In case they fail you have a broken `main`.

So, my suggestion would be to work on the integration tests and unit tests to run below 10 minutes, by focusing less on system tests. 

System tests are then executed on an regular basis, but not after every merge.

Reply:
> In my opinion, running tests after the change has been merged is cumbersome. In case they fail you have a broken main.

This was not my proposal. We need to run the tests before merging but after everything else was approved on. Which means, in the ideal case we have a bunch of revisions on the PR, hit `bors r+` and then the slow tests are run. If they fail (which they shouldn't because you tested it locally already), then we merge into master.